reactor-netty-http/src/main/java/reactor/netty/http/client/AbstractHttpClientMetricsHandler.java

```diff
@@ -76,64 +80,88 @@ protected AbstractHttpClientMetricsHandler(AbstractHttpClientMetricsHandler copy
 	@Override
 	@SuppressWarnings("FutureReturnValueIgnored")
 	public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise) {
-		if (msg instanceof HttpRequest) {
-			method = ((HttpRequest) msg).method().name();
-
-			ChannelOperations<?, ?> channelOps = ChannelOperations.get(ctx.channel());
-			if (channelOps instanceof HttpClientOperations) {
-				HttpClientOperations ops = (HttpClientOperations) channelOps;
-				path = uriTagValue == null ? ops.path : uriTagValue.apply(ops.path);
-				contextView = ops.currentContextView();
+		try {
+			if (msg instanceof HttpRequest) {
+				extractDetailsFromHttpRequest(ctx, (HttpRequest) msg);
 			}
 
-			dataSentTime = System.nanoTime();
-		}
-
-		if (msg instanceof ByteBufHolder) {
-			dataSent += ((ByteBufHolder) msg).content().readableBytes();
-		}
-		else if (msg instanceof ByteBuf) {
-			dataSent += ((ByteBuf) msg).readableBytes();
-		}
-
-		if (msg instanceof LastHttpContent) {
-			SocketAddress address = ctx.channel().remoteAddress();
-			promise.addListener(future -> recordWrite(address));
+			dataSent += extractProcessedDataFromBuffer(msg);
+
+			if (msg instanceof LastHttpContent) {
+				SocketAddress address = ctx.channel().remoteAddress();
+				promise.addListener(future -> {
+					try {
+						recordWrite(address);
+					} catch (RuntimeException e) {
+						log.warn("Exception caught while recording metrics.", e);
+						// Allow request-response exchange to continue, unaffected by metrics problem
+					}
+				});
+			}
+		} catch (RuntimeException e) {
+			log.warn("Exception caught while recording metrics.", e);
+			// Allow request-response exchange to continue, unaffected by metrics problem
```

Comment:
Do we need this `try/catch`? Is it possible to have a `RuntimeException` not related to metrics?

Reply:
No, this is not necessary. I likely introduced this before I found that the actual recording is done in a listener here.

Reply:
Actually we do need here this because `uriTagValue` is provided by the user and may throw an exception - see the failing CI build

Reply:
Added this back